rwengine/src/engine/SaveGame.cpp

```diff
@@ -1322,9 +1322,39 @@ bool SaveGame::getSaveInfo(const std::string& file, BasicState* basicState) {
     return true;
 }
 
+#ifdef RW_WINDOWS
+char* readUserPath() {
+    LONG retval;
+    const char* lpSubKey = "Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Folders";
+    const char* lpValueName = "Personal";
+    DWORD lpType = REG_SZ;
+    HKEY phkResult;
+    static char lpData[1000];
+    DWORD lpcbData = sizeof(lpData);
+
+    retval = RegOpenKeyEx(HKEY_CURRENT_USER, lpSubKey, 0, KEY_READ, &phkResult);
+    if (ERROR_SUCCESS != retval) { goto exit; }
+
+    retval = RegQueryValueEx(phkResult, lpValueName, NULL, &lpType, (LPBYTE) lpData, &lpcbData);
+    if (ERROR_SUCCESS != retval) { goto exit; }
```

Comment:
This goto jumps over the RegCloseKey call. I think it's clearer to use return when exiting from small functions.